### PR TITLE
Added lower() and upper() functions based on trim function

### PR DESCRIPTION
**Pull request by mirimmar**
Signed-off-by: lukasz palczewski <palczewskilukasz@tutanota.com>

### CODE REVIEW

**Comment by rbock** [comment]
Thanks for adding these functions. However, I think your commit is based on a rather old version of sqlpp11?

**Comment by mirimmar** [comment]
> 

I think I created the fork from the master branch, so if the master is old, than yes...

**Comment by rbock** [comment]
I suspect you created the fork off the master branch which was replaced by the main branch quite a while ago.
A lot of repositories have made this transition.

In any case, the `serializer_t` template was replaced by a `serialize` function quite a while ago. No worries, it is a very simple change.

**Comment by mirimmar** [comment]
Ok. I have changed it.

File: include/sqlpp11/lower.h [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+    context << "LOWER(";
+    serialize(t._expr, context);
+    context << ")";
+    return context;
+  }
+
+  template <typename T>
+  auto lower(T t) -> lower_t<noop, wrap_operand_t<T>>
+  {
+    static_assert(is_expression_t<wrap_operand_t<T>>::value, "lower() requires an expression as argument");
```
Comment (rbock) [inline include/sqlpp11/lower.h:67]: I think it would make sense to add a static assert for `is_text_t`?

**Comment by rbock** [comment]
Great, thanks!

Added one small comment. Otherwise, I think this looks ready.

File: include/sqlpp11/lower.h (hunk 1 above)
Comment (heatherlorry94com) [reply include/sqlpp11/lower.h:67]: Yea I think so

**Comment by rbock** [comment]
Superseded by [3fba3e7](https://github.com/rbock/sqlpp11/commit/3fba3e721e5fb4df16cbfb6f598e27e2505c65ca)

**Pull request closed without merging**